searx/engines/sqlite.py

```diff
@@ -0,0 +1,49 @@
+# SPDX-License-Identifier: AGPL-3.0-or-later
+# lint: pylint
+# pylint: disable=missing-function-docstring
+
+"""SQLite database (Offline)
+
+"""
+
+import sqlite3
+
+engine_type = 'offline'
+database = ""
+query_str = ""
+limit = 10
+paging = True
+result_template = 'key-value.html'
+
+def init(engine_settings):
+    if 'query_str' not in engine_settings:
+        raise ValueError('query_str cannot be empty')
+
+    if not engine_settings['query_str'].lower().startswith('select '):
+        raise ValueError('only SELECT query is supported')
+
+def search(query, params):
+    query_params = {'query': query}
+    query_to_run = query_str + ' LIMIT {0} OFFSET {1}'.format(limit, (params['pageno'] - 1) * limit)
```

Comment:
To avoid any `.format` (??):
```python
query_params = {'query': query, 'limit': limit, 'offset': (params['pageno'] - 1) * limit}
query_to_run = query_str + ' LIMIT :limit OFFSET :offset'
```

(not tested)

Reply:
> To avoid any .format (??)

LGTM, thanks! / BTW: using _execute parameters_ `cur.execute(..., query_params)` is more safe from SQL injection!